### ISSUE TITLE
Blob Service Example fails with XML parse error

### ISSUE THREAD

**Issue by joverwey** [Azure/azure-sdk-for-rust#1366]
I am trying to run the this example: `services\svc\blobstorage\examples\list_containers.rs` but I am getting an error:
`page error: failed to deserialize the following xml into a azure_svc_blobstorage::package_2021_12::models::ListContainersSegmentResponse`
```

<?xml version="1.0" encoding="utf-8"?>
<EnumerationResults ServiceEndpoint="https:///XXXX.blob.core.windows.net/">
    <Containers>
        <Container>
            <Name>azureml</Name>
            <Properties>
                <Last-Modified>Fri, 08 Sep 2023 16:36:39 GMT</Last-Modified>
                <Etag>"XXXX"</Etag>
                <LeaseStatus>unlocked</LeaseStatus>
                <LeaseState>available</LeaseState>
                <DefaultEncryptionScope>$account-encryption-key</DefaultEncryptionScope>
                <DenyEncryptionScopeOverride>false</DenyEncryptionScopeOverride>
                <HasImmutabilityPolicy>false</HasImmutabilityPolicy>
                <HasLegalHold>false</HasLegalHold>
                <ImmutableStorageWithVersioningEnabled>false</ImmutableStorageWithVersioningEnabled>
            </Properties>
        </Container>
        <Container>
            <Name>azureml-blobstore-XXXX</Name>
            <Properties>
                <Last-Modified>Fri, 08 Sep 2023 16:36:39 GMT</Last-Modified>
                <Etag>"XXXX"</Etag>
                <LeaseStatus>unlocked</LeaseStatus>
                <LeaseState>available</LeaseState>
                <DefaultEncryptionScope>$account-encryption-key</DefaultEncryptionScope>
                <DenyEncryptionScopeOverride>false</DenyEncryptionScopeOverride>
                <HasImmutabilityPolicy>false</HasImmutabilityPolicy>
                <HasLegalHold>false</HasLegalHold>
                <ImmutableStorageWithVersioningEnabled>false</ImmutableStorageWithVersioningEnabled>
            </Properties>
        </Container>
    </Containers>
    <NextMarker />
</EnumerationResults>
```

I know that I could use the blob storage API instead, but I am getting the same error using the filestorage generated API, which is actually what I am interested in since it does not have an SDK equivalent.

**Comment by demoray** [comment]
I can replicate your issue.  Investigating.

**Comment by demoray** [comment]
In order to fully fix the example, the fixes within #1381 and #1383 need to be merged and the crates regenerated.
